core/image/astc/astc.cc

```diff
@@ -13,77 +13,109 @@
 // limitations under the License.
 
 #include <stdio.h>
+#include <string.h>
 
 #include "astc.h"
+#include "third_party/astc-encoder/Source/astcenccli_internal.h"
 
-#include "third_party/astc-encoder/Source/astc_codec_internals.h"
+int init_astc_for_decode(astcenc_profile profile,
```

Comment:
Instead of having a go callback that just ends up printing to stdout, return the actual error code and do the string handling in the go side to turn it into proper go errors. I think you can reference the enum values via `C.<enum name>` and should be able to call the `astcenc_get_error_string` function directly via `C.astcenc_get_error_string`.

Reply:
Makes sense

Reply:
I have just added a wrapper function as we have to expose the astcenc_get_error_string function to be able to call from Go side.

I think that rather than exposing the direct astcenc function and entangle the astc library with go code, it is better to decouple them with a wrapper in between.